src/jit/flowgraph.cpp

```diff
@@ -4857,7 +4859,8 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, BYTE*
                     fgObserveInlineConstants(opcode, pushedStack, isInlining);
                 }
                 break;
-
+            case CEE_RET:
+                retBlocks++;
```

Comment:
Yeah, there has to be a `ret` after a `tail.call`. If there's no `ret` then the JIT fails to compile the method so we don't need to be concerned about what happens if someone feeds the JIT IL code that fails to conform to the ECMA spec. An invalid method would be flagged as "no return" and that's OK, calling such a method would result in an "InvalidProgramException" being thrown.

So counting `CEE_RET` and `CEE_JMP` is enough, if there aren't any then the method doesn't return (either because it throws or because it contains an infinite loop).